File: app/src/main/java/com/example/marco/weather/Locations/LocationsViewModel.java
package com.example.marco.weather.Locations;

import com.example.marco.weather.Data.Location;
import com.example.marco.weather.Tool.RealmStorage;
import com.example.marco.weather.Tool.Storage;

import java.util.List;

class LocationsViewModel {
    private Storage storage;

    LocationsViewModel () {
        storage = new RealmStorage();
    }

    List<Location> getLocations() {
        return storage.getLocations();
    }

    String deleteLocation(int position) {
        Location location = storage.getLocation(position);
        String name = location.getName();
        storage.removeLocation(location.getId());
        return name;
    }

    String getWeather(int position) {
        return storage.getLocation(position).getForecast(0).toString();
    }
}

File: app/src/main/java/com/example/marco/weather/Locations/LocationsViewModel.kt
package com.example.marco.weather.Locations

import com.example.marco.weather.Data.Location
import com.example.marco.weather.Tool.RealmStorage

internal class LocationsViewModel {
    private val storage = RealmStorage()

    val locations: List<Location>
        get() = storage.locations

    fun deleteLocation(position: Int): String {
        val name = storage.getLocation(position).name
        storage.removeLocation(storage.getLocation(position).id)
        return name
    }

    fun getWeather(position: Int): String {
        return storage.getLocation(position).getForecastAt(0).toString()
    }
}
